kubernetes/linux/setup.sh

```diff
@@ -9,88 +9,68 @@ else
     ARCH=$1
 fi
 
-#Download utf-8 encoding capability on the ama-logs container.
-#upgrade apt to latest version
-apt-get update && apt-get install -y apt && DEBIAN_FRONTEND=noninteractive apt-get install -y locales
+sudo tdnf install ca-certificates-microsoft -y
+sudo update-ca-trust
 
+sudo tdnf install ruby-3.1.3 -y
+rm /usr/lib/ruby/gems/3.1.0/specifications/default/openssl-3.0.1.gemspec
+rm -rf /usr/lib/ruby/gems/3.1.0/gems/openssl-3.0.1
```

Comment:
you would need to remove the rvm and find gems as well. please refer the ci_prod branch for the latest change.